### DISCUSSION TITLE
Programmatically pan the map

### DISCUSSION THREAD

**Discussion by Sprechen** [maplibre/maplibre-native#2088] [Q&A]
I'm using maplibre for Android.
In my app, users can control the map through a bluetooth remote controller, that has a small arrow pad.
I need to programmatically pan the map when the user press the corresponding key.
I don't found any directly method to ask the map view to pan to any direction, simulating an user "pan" gesture.
I would like NOT to use a "flyTo" or similar method, since this means that I need to calculate a target location, that could be quite complicated (e.g. if I need to pan left, I also need to know the orientation of the user, in order to correctly calculate the coordinates on his left).

How can I achive this behaviour?

Thanks!

**Comment by acalcutt** [comment]
I don't have an answer for you in Android, but in maplibre-gl-js it sounds like you want something like this example right?
https://maplibre.org/maplibre-gl-js/docs/examples/game-controls/

**Reply by acalcutt** [reply]
Does maplibre have MoveBy? that seems like the equivilent of PanBy in the js example. https://docs.mapbox.com/android/maps/api/10.0.0-beta.12/-mapbox%20-maps%20-android/com.mapbox.maps.plugin.animation/move-by.html . 

I tried to find it in maplibre docs, but I honestly couldn't even find flyto

**Reply by stefankarschti** [reply]
@acalcutt  is correct. 
@Sprechen I think what you're looking for is `void Map::moveBy(const ScreenCoordinate& point, const AnimationOptions& animation)` https://github.com/maplibre/maplibre-native/blob/14bfb5df7257848ee0d14570e349a07996c9469f/include/mbgl/map/map.hpp#L70

Example usage: https://github.com/maplibre/maplibre-native/blob/14bfb5df7257848ee0d14570e349a07996c9469f/platform/ios/src/MLNMapView.mm#L2089C1-L2128C1

**Reply by Sprechen** [reply]
well, I also found the method `public void scrollBy(float x, float y, long duration)` that should let me achive the same behaviour, without installing any additional plugin.
I haven't tested it yet, but it seems interesting!
